reaching questions from backend and displaying it in question-list

athavy92 opened the pull request:
Used Input() to display questions from the backend in the question-list component and each question is its own preview-question component

athavy92 committed:
reaching questions from backend and displaying it in question-list component

AdamDKing reviewed (changes requested):
Rework this PR to use "Question" instead of "Post" in all cases, and move functionality out of post-form-component and posts-component.  We're removing those because they overlap  (in fact, we already have consolidated them on dev, if you want to merge that in)

athavy92 committed:
refactored post to question

Renamed all "post" to "question" for uniformity.
Also, created a similar question.service.ts as the there are several components that use post.service.ts that would be broken if I changed it.

AdamDKing reviewed (changes requested):
As discussed, we can't lightly rewrite classes tat other people are using.  Work with the existing Question.ts, potentially adding properties.  It's OK to make it an interface.

athavy92 committed:
added a few properties in Question.ts model

Added a few properties that are needed in the existing Question.ts file in the models directory. Ensured added changes didn't break any functionality.

athavy92 committed:
merge with dev

athavy92 committed:
preview-question to view-question component

Added functionality to go from the Questions List component to view the View Question component to display the full question with responses.

athavy92 committed:
changed fields

AdamDKing committed:
merge dev before PR

AdamDKing reviewed (approved):
Approving and finishing cleaning/standardizing on another branch off of dev
